src/Motor.Extensions.Hosting.RabbitMQ/Options/RabbitMQBaseOptions.cs

```diff
@@ -1,22 +1,36 @@
 using System;
+using System.Net.Security;
+using System.Security.Authentication;
 using Motor.Extensions.Hosting.RabbitMQ.Validation;
 
 namespace Motor.Extensions.Hosting.RabbitMQ.Options;
 
+public sealed record RabbitMQSslOptions
+{
+    public bool Enabled { get; init; } = false;
+    public SslProtocols Protocols { get; init; } = SslProtocols.Tls12;
+
+    public string ClientCertPath { get; init; } = string.Empty;
+    public string ClientCertPassword { get; init; } = string.Empty;
+
+    public RemoteCertificateValidationCallback? CertificateValidationCallback { get; set; }
```

Comment:
Do we need this? I don't think there's a benefit in having an option which is never even configured.

Reply:
I would probably keep this. Being able to validate a certificate, e.g., with certificate pinning is a reasonable use case when using TLS. Also it doesn't really hurt to provide this option, does it?

Reply:
I was just wondering because RabbitMQ's `SslOption` class supports quite a few options which sound reasonable to me but are not included.
But my actual concern was: Do we want to ship dead code just because we think that we might build a feature which uses it at some point in the future?

Reply:
It's not dead code.
It's true that you can't configure it from the appsettings.json but that doesn't mean you can't configure this very option type manually with setting this hook to change the default behavior of the cert validation.

Apart from that: which options are missing you'd like to include? There is another hook to select the client certificate for TLS authentication but I didn't include that one by purpose because I thought client authentication should probably be done in a second PR.
I check for other useful options if you like.

Reply:
I just had a quick look at the class and found an option for a certificate revocation check for example. I don't know if that's ever going to happen, but anyway: Why would we not want to have this ability?